### PR TITLE
246 register create storage account for temporary data

### PR DESCRIPTION
**Pull request by divesh987**
### Context

Created a toggle for creating the temp data storage account. 
Storage account should not be created for review apps so creation is toggled to off for review variables file

### Guidance to review
Can set ```deploy_temp_data_storage_account``` to true and deploy a review app to see that storage account gets created.

### Link to Trello Card
https://trello.com/c/Vy9zOfCL/246-register-create-storage-account-for-temporary-data

### Important business

- [ ] Does this PR introduce any PII fields that need to be overwritten or deleted in db/scripts/sanitise.sql?
- [ ] Does this PR change the database schema? If so, have you updated the config/analytics.yml file and considered whether you need to send 'import_entity' events?
- [ ] Do we need to send any updates to DQT as part of the work in this PR?
- [ ] Does this PR need an ADR?

NB: Please notify the #twd_data_insights team and ask for a review if new fields are being added to analytics.yml

### CODE REVIEW

**Comment by divesh987** [comment]
[Register - Create storage account for temporary data](https://trello.com/c/Vy9zOfCL/246-register-create-storage-account-for-temporary-data)

File: terraform/aks/variables.tf [hunk 1]
```diff
@@ -250,3 +250,5 @@ variable azure_tempdata_storage_account_name {}
 variable azure_storage_account_replication_type { default = "LRS"}
 
 variable azure_region_name { default = "uk south" }
+
+variable "deploy_temp_data_storage_account" {default = true }
```
Comment (smneal) [inline terraform/aks/variables.tf:254]: `{default = true }` to `{ default = true }` for consistency. Maybe a `terraform fmt` or the VS Code plugin?

File: terraform/modules/azure/variables.tf [hunk 2]
```diff
@@ -26,3 +26,5 @@ variable "region_name" {
 locals {
   hosting_environment = var.environment_name
 }
+
+variable "deploy_temp_data_storage_account" {default = true }
```
Comment (smneal) [inline terraform/modules/azure/variables.tf:30]: `{default = true }` to `{ default = true }` for consistency. Maybe a `terraform fmt` or the VS Code plugin?

**Review by smneal — COMMENTED** [review]
Just a few comments for review 👌🏻

**Review by smneal — COMMENTED** [review]
I assume this will also need adding to the `review_aks.tfvars.json` file as well

File: terraform/aks/workspace-variables/dv_review_aks.tfvars.json [hunk 3]
```diff
@@ -25,5 +25,6 @@
       "replicas": 1,
       "memory_max": "1Gi"
     }
-  }
+  },
+  "deploy_temp_data_storage_account": false
```
Comment (RMcVelia) [inline terraform/aks/workspace-variables/dv_review_aks.tfvars.json:29]: I think azure_tempdata_storage_account_name and azure_resource_group_name become irrelevant if deploy_temp_data_storage_account is false, but deploy-plan fails if you remove them.

**Comment by divesh987** [comment]
> I assume this will also need adding to the `review_aks.tfvars.json` file as well

Added!

File: terraform/aks/workspace-variables/dv_review_aks.tfvars.json (hunk 3 above)
Comment (divesh987) [reply terraform/aks/workspace-variables/dv_review_aks.tfvars.json:29]: I just tested by removing those params and yes deploy-plan prompts you to fill them in so might be best to just keep them in?

Comment (RMcVelia) [reply terraform/aks/workspace-variables/dv_review_aks.tfvars.json:29]: preference would be to not have to specify vars we don't need

**Pull request merged**